module/os/linux/zfs/zfs_ctldir.c

```diff
@@ -807,7 +808,9 @@ zfsctl_root_lookup(struct inode *dip, const char *name, struct inode **ipp,
 	if ((error = zfs_enter(zfsvfs, FTAG)) != 0)
 		return (error);
 
-	if (strcmp(name, "..") == 0) {
+	if (zfsvfs->z_show_ctldir == ZFS_SNAPDIR_DISABLED) {
+		error = SET_ERROR(ENOENT);
```

Comment:
It feels like `*ipp` might be uninitialized in this case, and doing `*ipp = NULL;` instead would give the same effect, but cleaner.

Reply:
Good idea, that nicely matches the `else` block too.